contracts/utils/math/Math.sol

```diff
@@ -68,11 +73,38 @@ library Math {
      */
     function tryMod(uint256 a, uint256 b) internal pure returns (bool success, uint256 result) {
         unchecked {
-            if (b == 0) return (false, 0);
-            return (true, a % b);
+            success = b > 0;
+            assembly ("memory-safe") {
+                // In EVM a value modulus zero is equal to zero.
+                result := mod(a, b)
+            }
         }
     }
 
+    /**
+     * @dev Unsigned saturating addition, bounds to `2 ** 256 - 1` instead of overflowing.
+     */
+    function saturatingAdd(uint256 a, uint256 b) internal pure returns (uint256) {
+        (bool success, uint256 result) = tryAdd(a, b);
+        return ternary(success, result, type(uint256).max);
+    }
+
+    /**
+     * @dev Unsigned saturating subtraction, bounds to zero instead of overflowing.
```

Comment:
```suggestion
     * @dev Unsigned saturating subtraction, bounds to zero instead of underflowing.
```

Reply:
Muy understanding is that both `type(uint256).max + 1` and `type(uint256).min - 1` are overflow. An underflow is related to precision loss in floating point representation. So I'd say this is fine

Reply:
Ah seems you are correct. Thanks!